activerecord/lib/active_record/tasks/sqlite_database_tasks.rb

```diff
@@ -49,6 +49,9 @@ def structure_dump(filename, extra_flags)
 
         ignore_tables = ActiveRecord::SchemaDumper.ignore_tables
         if ignore_tables.any?
+          if ignore_tables.any?(Regexp)
```

Comment:
I don't think you need to check for `Regexp` presence, `String#===` will work just as well.

Removing it would make the code flow a bit better IMHO.

Reply:
That was done to avoid extra db query for tables when there are no regexps. Premature optimization and better to remove?

Reply:
Yeah, I don't think it's worth the more contrived code.

Reply:
Changed.